internal/testrunner/runners/system/runner.go

```diff
@@ -213,16 +214,33 @@ func (r *runner) run() error {
 				Total struct {
 					Value int
 				}
+				Hits []struct {
+					Source Document `json:"_source"`
+				}
 			}
 		}
 
 		if err := json.NewDecoder(resp.Body).Decode(&results); err != nil {
 			return false, errors.Wrap(err, "could not decode search results response")
 		}
 
-		hits := results.Hits.Total.Value
-		logger.Debugf("found %d hits in %s data stream", hits, dataStream)
-		return hits > 0, nil
+		numHits := results.Hits.Total.Value
+		logger.Debugf("found %d hits in %s data stream", numHits, dataStream)
+		if numHits == 0 {
+			return false, nil
+		}
+
+		var multiErr multierror.Error
+		for _, hit := range results.Hits.Hits {
+			if hit.Source.Error != nil {
+				multiErr = append(multiErr, errors.New(hit.Source.Error.Message))
+			}
+		}
+
+		if len(multiErr) > 0 {
+			return false, multiErr
```

Comment:
Should we wrap `multiErr` with some overall message so it's not just a collection of error messages from ES?

Reply:
I like this idea. Fixed.